fiaas_deploy_daemon/extension_hook_caller.py

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env python
+# -*- coding: utf-8
+
+# Copyright 2017-2021 The FIAAS Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import json
+import logging
+from collections import OrderedDict
+
+from six import string_types
+
+
+LOG = logging.getLogger(__name__)
+
+
+class ExtensionHookCaller(object):
+
+    def __init__(self, config, session):
+        self._url = config.hook_service
+        self._session = session
+
+    def apply(self, kind, obj, app_spec):
+        if self._url is None:
+            return obj
+        url = str(self._url) + "/fiaas/deploy/" + str(kind)
+        dump = json.dumps({"object": obj.as_dict(), "application": self._app_spec_to_dict(app_spec)})
+        response = self._session.post(
+            url,
+            data=dump,
+            headers={'Content-Type': 'application/json', 'Accept': 'application/json'}
+        )
+        if response.status_code == 200:
+            data = response.json()
+            obj.update_from_dict(data)
+        elif response.status_code != 404:
+            raise Exception("The api call to " + url + " returns an status code of " + response.status_code)
```

Comment:
In general, I would prefer specific exceptions rather than a generic `Exception`. In this case, you already have one available, by calling `resp.raise_for_status()`.

Reply:
Fixed